### PR TITLE
feat: adds image adding option to all questions

### LINKED ISSUE #1258: [FEATURE][FormTribe 🔥][750 Points] Add Image to every question
### Welcome to the FormTribe

This issue will get you the points mentioned in issue title for the [FormTribe Hackathon](https://formtribe.com)

**Preparation:**
If this is your first time contributing, please read:

- Our [introductory blog post](https://formbricks.com/blog/join-the-formtribe)
- Overview of [Code Guidelines at Formbricks.](https://formbricks.com/docs/contributing/how-we-code)

**Only when your PR follows _all_ of the best practices it will get merged.**

Want more points quick? Complete [Side Quests 🏰 ](https://formbricks.com/formtribe#side-quests)

--------

### Is your feature request related to a problem? Please describe.

Users want to add images and videos to have a richer context.

### Describe the solution you'd like

Allow adding images to every question (video will follow).

### Note
This PR will touch a lot of files because you have to embed the image adding and displaying functionality for almost every question type.

**Please only do it for 1 question type at first, ship it and get it approved! Otherwise you will have to deal with a loooot of merge conflicts all the time :)**

### Tasks

1. At the end of the question input field, add Lucide icon 'image-plus'
image

Note: This can be a bit hacky for now, placing it there with absolute positioning. We are rewriting the question input field into a custom component for the Recall information functionality anyways. So we will refactor this anyways :)

------

2. On click, it adds this upload window above the question input field: 
image

------

3. On drag (or click) the native file manager of the client opens and the user can add an image. Please reuse the File Upload component used in the Welcome Card 🙏

------

4. When the image is added, it looks like this:
image

------

5. Create a new reusable component to update the question types in the survey editor to display the images:

**Link survey:**
image

**In app survey:**
image

------

If question come up, please reach out in Discord! :)

### Thanks!

### PR DESCRIPTION
**Pull request by rtpa25**
## What does this PR do?

Fixes #1258 

https://www.loom.com/share/6cb2314656094e5e90f0b2d5abc71ca8

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- StoryBook tests 

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/CTAQuestionForm.tsx [hunk 1]
```diff
@@ -24,22 +26,42 @@ export default function CTAQuestionForm({
...
+              value={question.headline}
+              onChange={(e) => updateQuestion(questionIdx, { headline: e.target.value })}
+              isInvalid={isInValid && question.headline.trim() === ""}
+            />
+            <ImagePlusIcon
+              aria-label="Toggle image uploader"
+              className="ml-2 h-4 w-4 cursor-pointer text-slate-400 hover:text-slate-500"
+              onClick={() => setShowImageUploader((prev) => !prev)}
+            />
+          </div>
```
Comment (rtpa25) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/CTAQuestionForm.tsx:64]: Refactor into a separate component

File: apps/web/package.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
     "lru-cache": "^10.0.1",
     "lucide-react": "^0.287.0",
     "mime": "^3.0.0",
-    "next": "13.5.5",
+    "next": "13.5.6",
```
Comment (rtpa25) [inline apps/web/package.json:40]: https://github.com/vercel/next.js/issues/56947 <Reason for patch upgrade>

**Comment by rtpa25** [comment]
LGTM

**Comment by rtpa25** [comment]
@anjy7 Not necessarily, UI components are Input, Label etc which do not have any business specific logic, but this has packages/ui is the ui library of a product, this does not belong there IMO thoughts @mattinannt

**Review by neilchauhan2 — CHANGES_REQUESTED** [review]
@rtpa25 Thank you so much for your contribution 🎉. The functionality is working great, but I have mentioned a few issues, could you please address them. I'll review this PR again post that.

cc: @mattinannt

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -17701,6 +17786,45 @@ packages:
...
+    peerDependenciesMeta:
+      '@opentelemetry/api':
+        optional: true
+      sass:
+        optional: true
+    dependencies:
+      '@next/env': 13.5.6
+      '@swc/helpers': 0.5.2
+      busboy: 1.6.0
+      caniuse-lite: 1.0.30001547
```
[the remark this answers is not in the document]
Comment (rtpa25) [reply pnpm-lock.yaml:17807]: Devs don't change lock files, I have just upgraded the nextjs version as mentioned here https://github.com/formbricks/formbricks/pull/1305#discussion_r1364823631. These are peer deps for that, I have no control over that

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -28,7 +28,7 @@ importers:
         version: 3.13.0
       turbo:
         specifier: latest
-        version: 1.10.14
+        version: 1.10.12
```
[the remark this answers is not in the document]
Comment (rtpa25) [reply pnpm-lock.yaml:31]: as I said below

File: packages/surveys/src/components/MultipleChoiceMultiQuestion.tsx [hunk 5]
```diff
@@ -93,6 +93,12 @@ export default function MultipleChoiceSingleQuestion({
         onSubmit({ [question.id]: newValue });
       }}
       className="w-full">
+      {question.imageUrl && (
+        <div className="my-4 rounded-md">
+          {/* eslint-disable-next-line @next/next/no-img-element */}
+          <img src={question.imageUrl} alt="question-image" className={"my-4 rounded-md"} />
+        </div>
+      )}
```
[the remark this answers is not in the document]
Comment (rtpa25) [reply packages/surveys/src/components/MultipleChoiceMultiQuestion.tsx:101]: The es-lint rule is wrong hence warning too, this package is written with preact, can't access next image here. Can in dev cause of turbo common node_modules, but not in prod

File: packages/surveys/src/components/ConsentQuestion.tsx [hunk 6]
```diff
@@ -28,6 +28,12 @@ export default function ConsentQuestion({
 }: ConsentQuestionProps) {
   return (
     <div>
+      {question.imageUrl && (
+        <div className="my-4 rounded-md">
+          {/* eslint-disable-next-line @next/next/no-img-element */}
+          <img src={question.imageUrl} alt="question-image" className={"my-4 rounded-md"} />
+        </div>
```
[the remark this answers is not in the document]
Comment (rtpa25) [reply packages/surveys/src/components/ConsentQuestion.tsx:35]: check comment below

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionFormInput.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            id="headline"
+            name="headline"
+            value={question.headline}
+            onChange={(e) => updateQuestion(questionIdx, { headline: e.target.value })}
+            isInvalid={isInValid && question.headline.trim() === ""}
+          />
+          <ImagePlusIcon
+            aria-label="Toggle image uploader"
+            className="ml-2 h-4 w-4 cursor-pointer text-slate-400 hover:text-slate-500"
+            onClick={() => setShowImageUploader((prev) => !prev)}
```
[the remark this answers is not in the document]
Comment (rtpa25) [reply apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionFormInput.tsx:56]: this is more readable as it's a one liner, and is repeated in a ton of places inside the codebase, creating separate handlers is unnecessary clutter and over head cause you need to give good names to them too

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionFormInput.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    <div className="mt-3">
+      <Label htmlFor="headline">Question</Label>
+      <div className="mt-2 flex flex-col gap-6">
+        {showImageUploader && (
+          <FileInput
+            allowedFileExtensions={["png", "jpeg", "jpg"]}
+            environmentId={environmentId}
+            onFileUpload={(url: string) => {
+              updateQuestion(questionIdx, { imageUrl: url });
+            }}
```
[the remark this answers is not in the document]
Comment (rtpa25) [reply apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionFormInput.tsx:39]: Look at other places updateQuestion is being used and this https://github.com/formbricks/formbricks/pull/1305#discussion_r1365537336

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/MultipleChoiceMultiForm.tsx [hunk 9]
```diff
@@ -155,21 +159,18 @@ export default function MultipleChoiceMultiForm({
...
-            value={question.headline}
-            onChange={(e) => updateQuestion(questionIdx, { headline: e.target.value })}
-            isInvalid={isInValid && question.headline.trim() === ""}
-          />
-        </div>
-      </div>
+      <QuestionFormInput
+        environmentId={environmentId}
+        isInValid={isInValid}
+        ref={questionRef}
```
[the remark this answers is not in the document]
Comment (rtpa25) [reply apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/MultipleChoiceMultiForm.tsx:169]: it routes back to the Input inside the questionforminput component as used above

**Review by neilchauhan2 — APPROVED** [review]

**Comment by neilchauhan2** [comment]
@mattinannt done with the review, could you please look into this.

**Pull request merged**